PyRDF/Backends/Dist.py

```diff
@@ -680,6 +235,13 @@ def distribute_unique_paths(self, paths):
         """
         pass
 
+    def optimize_npartitions(self, npartitions):
```

Comment:
Is this new? What's its purpose?

Reply:
Kind of. Previously in `SparkBackend` there was the `get_partitions` method. This was actually trying to guess the right amount of partitions for that particular instance of the backend. So I renamed it to `optimize_npartitions` and also made it part of the parent class interface so this can be subclassed. This is in preparation for future backends where we will want to have a way of "optimizing" the partitions parameter of the single dataframe.

Reply:
The logic is also backend-dependent, following the loose rule that the number of partitions should be at greater or equal to  the number of available cores on the cluster. In the spark implementation this is `spark.executor.instances` * `spark.executor.cores`

Reply:
So this is invoked by the backends when the user does not specify the number of partitions?

Reply:
Correct. The number of partitions is set to a default value in the factory function that creates the dataframe (currently 2). Then each backend can change this default value to something more clever